### PR TITLE
Only update if actively refreshing

### PR DESCRIPTION
**Pull request by crobibero**
Fixes https://github.com/jellyfin/jellyfin/issues/11264

`.AddOrUpdate()` is doing essentially the same thing internally, this just removes the exception

### CODE REVIEW

**Review by mark-monteiro — APPROVED** [review]

**Review by Shadowghost — APPROVED** [review]

**Comment by nielsvanvelzen** [comment]
Even with the exception gone, the actual issue where it tries to update out of order still exists. I believe this affects the scanner progress in the dashboard. Is that fine or do we want to fix it in a different way?

**Comment by crobibero** [comment]
I think the issue is that the refresh progress is called after refresh completes. When the refresh completes and the item is removed from the dictionary a progress of `100` is sent, so I don't think this will have any negative impact on user experience

File: MediaBrowser.Providers/Manager/ProviderManager.cs [hunk 1]
```diff
@@ -969,7 +969,8 @@ public void OnRefreshProgress(BaseItem item, double progress)
             _logger.LogDebug("OnRefreshProgress {Id:N} {Progress}", id, progress);
 
             if (!_activeRefreshes.TryGetValue(id, out var current)
-                || !_activeRefreshes.TryUpdate(id, progress, current))
+                || progress <= current
```
Comment (mark-monteiro) [inline MediaBrowser.Providers/Manager/ProviderManager.cs:972]:
```suggestion
```
No need to check this condition twice?

File: MediaBrowser.Providers/Manager/ProviderManager.cs [hunk 2]
```diff
@@ -969,7 +969,8 @@ public void OnRefreshProgress(BaseItem item, double progress)
...
+                || (progress > current && !_activeRefreshes.TryUpdate(id, progress, current)))
             {
                 // Item isn't currently refreshing so don't trigger event.
```
Comment (mark-monteiro) [inline MediaBrowser.Providers/Manager/ProviderManager.cs:975]:
```suggestion
                // Item isn't currently refreshing, or update was received out-of-order, so don't trigger event.
```

**Review by mark-monteiro — APPROVED** [review]

**Comment by barronpm** [comment]
There should be a try-catch somewhere in the callers that can be removed now, if I understand the changes correctly, I added it when I was replacing our custom `IProgress` impl with `System.Progress`

**Review by nielsvanvelzen — APPROVED** [review]

**Pull request merged**